04-brute-force-and-exhaustive-search/195688/gyujin.java

```diff
@@ -0,0 +1,58 @@
+import java.io.*;
+import java.util.*;
+
+class Main {
+	private static int N;
+	private static String str;
+	private static ArrayList<String> list;
+	private static HashSet<String> set;
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		
+		N = Integer.parseInt(br.readLine());
+		str = br.readLine();
+		
+		set = new HashSet<>();      // 중복 제거하기 위해 HashSet 사용. 
+		
+		scoreBoard();               // P를 만들기 위함.
+        
+		list = new ArrayList<>(set);        // 중복을 제거한 HashSet을 정렬하기 위해 ArrayList에 담음.
+		Collections.sort(list);             // 오름차순 정렬.
+	
+		System.out.println(scoreMax());     // scoreBoard에서 했던거와 같이 scoreMax도 브루트포스로 검사하여 최대값을 return
+	}
+	
+	private static void scoreBoard() {          
```

Comment:
전역에 set을 선언하지 않고 HashSet을 반환하는 메서드로 만들어도 좋았을 것 같습니다!
```suggestion
	private static Set<String> scoreBoard() {
```

Reply:
대은님 말씀대로 전역에 set을 선언하지 않고 HashSet을 반환하는 메소드로 변경 후,
list에 바로 반환하니 코드가 더 깔끔해진 것 같습니다!
감사합니다!👍
코드 수정했습니다!🤝